xray/src/generation.rs

```diff
@@ -127,7 +128,27 @@ pub trait ColoringStrategy: Send {
     // the pixel (x, y) in the final tile image.
     fn get_pixel_color(&self, x: u32, y: u32, background_color: Color<u8>) -> Color<u8>;
 
-    fn attributes(&self) -> Vec<&'static str>;
+    fn attributes(&self) -> HashSet<String>;
+}
+
+trait BinnedColoringStrategy {
+    fn binning(&self) -> &Binning;
+    fn bins(&self, points_batch: &PointsBatch) -> Vec<i64> {
+        if let Some((attrib_name, size)) = self.binning() {
+            if let Some(attr_data) = points_batch.attributes.get(attrib_name) {
```

Comment:
q: don't we want to panic if the point cloud doesn't have `attrib_name` attribute?

Reply:
Or at least emit a warning

Reply:
I changed it to `expect` and a surrounding `match`.